[Gabriel Mario Antaputra] ip

thegrimbee opened the pull request:
# Kyrie

**TRY OUT KYRIE NOW!!!**  Kyrie is the leading text based chatbot in LaLaLand. It is:

- text-based
- easy to learn
- awesome

All you need to do is:

1. download it from [here](https://github.com/thegrimbee/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
   
And it is for the price of 0 DOLLARS!!!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

> [!WARNING]
> THIS CHATBOT IS SO GOOD YOU MIGHT NEVER USE ANY OTHER CHATBOT AGAIN

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

thegrimbee committed:
Changed name to kyrie and added welcome and goodbye message

thegrimbee committed:
added tasks and list command

thegrimbee committed:
added mark and unmark command'

thegrimbee committed:
Made Task class

thegrimbee committed:
added 3 types of task

thegrimbee committed:
Added tests

thegrimbee committed:
Added exceptions

thegrimbee committed:
Added delete and TaskList

thegrimbee committed:
Added TaskType

thegrimbee committed:
Update Kyrie to create data file if it does not exist yet

thegrimbee committed:
Update each task to have toFileString() method

thegrimbee committed:
Add DataHandler which handles the kyrie.txt file and update kyrie to load and save tasklist accordingly

thegrimbee committed:
Merge branch 'branch-level-7'

thegrimbee committed:
Stop tracking data folder

thegrimbee committed:
Stop tracking data

thegrimbee committed:
Add DateTime to handle dates and time and adjust the tasks accordingly

thegrimbee committed:
Merge branch 'branch-level-8'

thegrimbee committed:
Update DataHandler to Storage

thegrimbee committed:
Add Ui and basic template for other classes

thegrimbee committed:
Add parser and updated Kyrie to have main, run, and constructor

thegrimbee committed:
Add all commands

thegrimbee committed:
Merge branch 'branch-a-moreoop'

thegrimbee committed:
Move everything to kyrie package

thegrimbee committed:
Merge branch 'branch-a-packages'

thegrimbee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

thegrimbee committed:
Add junit tests for tasklist and todo

thegrimbee committed:
Merge branch 'branch-a-junit'

thegrimbee committed:
Add javadoc comments to all methods and classes

thegrimbee committed:
Add find command

thegrimbee committed:
Merge branch 'branch-a-javadoc'

thegrimbee committed:
Merge branch 'branch-level-9'

thegrimbee committed:
Add javadocs comment to level-9

nickt121 commented on src/main/java/kyrie/Command.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    /**
+     * Returns false as the default value for isExit.
```
Perhaps you might want to leave an empty line between the description and the parameters for javadocs in order to follow the javadocs guidelines more closely.

nickt121 commented on src/main/java/kyrie/Ui.java:
```diff
@@ -0,0 +1,136 @@
+     * Responds to the user.
+     * @param response The response to the user.
+     */
+    private void respond(String response) {
```
Perhaps using a more descriptive method name will be better.

nickt121 commented on src/main/java/kyrie/Task.java:
```diff
@@ -0,0 +1,68 @@
+     * @return True if the task is done, false otherwise.
+     */
+    public Boolean isDone() {
+        return this.isDone;
```
I like that you phrased boolean variable names in your code appropriately!

nickt121 commented on src/main/java/kyrie/Storage.java:
```diff
@@ -0,0 +1,106 @@
+    /**
+     * Creates the directory and file to save the task list to.
+     */
+    public void createDirAndFile() {
```
I like that your method names explains the method to the reader accurately and at a sufficient level of detail!

nickt121 commented on src/main/java/kyrie/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+ * Represents the task list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
I like how you clearly between single-valued and multi-valued variables. Good job!!

nickt121 reviewed (commented):
Very well done so far. Keep up the good work!!

thegrimbee committed:
fix based on feedback

qilstiano commented on src/main/java/kyrie/Parser.java:
```diff
@@ -0,0 +1,72 @@
+        String[] commandParts = commandString.split(" ");
+        String command = commandParts[0];
+        switch (command) {
+            case "bye":
```
Nice job on handling the switch statements! I missed out on this coding style requirement initially so it's awesome that you got it on your first try 👍

qilstiano commented on src/test/java/kyrie/TaskListTest.java:
```diff
@@ -0,0 +1,42 @@
+
+public class TaskListTest {
+    @Test
+    public void testTaskListConstructor() {
```
While it's not really related to coding style, I think this test for the max TaskList size was quite a good spot from you. Nice one! 👍

qilstiano commented on src/main/java/kyrie/DateTime.java:
```diff
@@ -0,0 +1,42 @@
+    private LocalDateTime dateTime;
+
+    /**
+     * Constructor for DateTime.
```
I believe this was mentioned by @nickt121 as well but leaving a line between params and the header would be more aligned with the coding style standards. While this format seems to be standardised throughout your project, perhaps you could consider this minor update in your future iterations?

qilstiano reviewed (commented):
Overall you've done a really solid job! Mostly just nits and really minor deviations from coding style. Excited to see what's to come for Kyrie in the future 💯

thegrimbee committed:
Add GUI to Kyrie

thegrimbee committed:
Fix inconsistent javafx version in build.gradle and fxml

thegrimbee committed:
Add assertions to validate input parameters in command classes

The assertions include:
*non null names and messages
*positive index for mark, unmark, and delete

thegrimbee committed:
Refactor code

The refactor includes:
*Renaming variables and fields
*Removing unused methods
*Standardising code

thegrimbee committed:
Merge pull request #2 from thegrimbee/branch-A-CodeQuality

Refactor code

thegrimbee committed:
Merge pull request #1 from thegrimbee/branch-A-Assertions

Add assertions to validate input parameters in command classes

thegrimbee committed:
Standardise kyrieexception message

thegrimbee committed:
Update parser and storage to follow SLAP

thegrimbee committed:
Add tag command and update storage and parser accordingly

thegrimbee committed:
Update commands to have command word to increase abstraction

thegrimbee committed:
Update error handling to handle more exceptions with clearer messages

thegrimbee committed:
Add new image for user

thegrimbee committed:
Update todo, deadline, and event to allow multiple word task descriptions

thegrimbee committed:
Add user guide and screenshot of application

thegrimbee committed:
Fix image path for github pages

thegrimbee committed:
Update README.md to have examples in each feature to standardise with the command summary

thegrimbee committed:
Fix README.md examples

thegrimbee committed:
Update README.md how to get started section

thegrimbee committed:
Update README.md to have datetime format

thegrimbee committed:
Update dialog-box to have darker colors

thegrimbee committed:
Update UI overall to have darker colors
